src/main/java/gift/product/service/WishService.java

```diff
@@ -2,46 +2,70 @@
 
 import gift.product.dto.LoginMember;
 import gift.product.dto.WishDto;
+import gift.product.model.Member;
+import gift.product.model.Product;
 import gift.product.model.Wish;
+import gift.product.repository.AuthRepository;
 import gift.product.repository.WishRepository;
 import java.util.List;
 import java.util.NoSuchElementException;
 import java.util.Optional;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.transaction.annotation.Transactional;
 
-@Transactional
+@Transactional(readOnly = true)
 public class WishService {
 
     private final WishRepository wishRepository;
     private final ProductService productService;
+    private final AuthRepository authRepository;
 
-    public WishService(WishRepository wishRepository, ProductService productService) {
+    public WishService(WishRepository wishRepository, ProductService productService,
+        AuthRepository authRepository) {
         this.wishRepository = wishRepository;
         this.productService = productService;
+        this.authRepository = authRepository;
     }
 
     public List<Wish> getWishAll(LoginMember loginMember) {
         return wishRepository.findAllByMemberId(loginMember.id());
     }
 
+    public Page<Wish> getWishAll(Pageable pageable) {
+        return wishRepository.findAll(pageable);
+    }
+
     public Wish getWish(Long id, LoginMember loginMember) {
         return getValidatedWish(id, loginMember);
     }
 
+    @Transactional
     public Wish insertWish(WishDto wishDto, LoginMember loginMember) {
-        productService.getProduct(wishDto.productId());
+        Product product = productService.getProduct(wishDto.productId());
 
-        Wish wish = new Wish(loginMember.id(), wishDto.productId());
-        wish = wishRepository.save(wish);
+        Member member = getMember(loginMember);
 
-        return wish;
+        Wish wish = new Wish(member, product);
+        return wishRepository.save(wish);
     }
 
+    @Transactional
     public void deleteWish(Long id, LoginMember loginMember) {
         getValidatedWish(id, loginMember);
         wishRepository.deleteById(id);
     }
 
+    private Member getMember(LoginMember loginMember) {
+        Optional<Member> member = authRepository.findById(loginMember.id());
+
+        if (member.isEmpty()) {
+            throw new NoSuchElementException("회원 정보가 존재하지 않습니다.");
+        }
+
+        return member.get();
+    }
+
```

Comment:
optional에는 `orElseThrow`와 `orElseGet` 이 존재하는데, 각각 가져온 객체가 empty상태일 때 예외를 바로 던지거나, 디폴트 값을 넣어줄 때 사용합니다!

여기서는 orElseThrow를 사용해 이렇게 리팩토링 할 수도 있곘네요!

```suggestion
private Member getMember(LoginMember loginMember) {
        return authRepository.findById(loginMember.id())
                .orElseThrow(() -> new NoSuchElementException("회원 정보가 존재하지 않습니다."));
    }
```

Reply:
꿀팁 감사합니다! 바로 적용하겠습니다.